coding-guides/using-audit-logs.md

```diff
@@ -0,0 +1,169 @@
+  if (!message.guild.me.hasPermission('MANAGE_CHANNELS') && !logs) { 
+    console.log('The logs channel does not exist and tried to create the channel but I am lacking permissions')
+  }
+  
+  // Now that we have the logs channel created...hopefully...we can send messages to it
+  // before we do that, lets establish who deleted the message
+  // The "type" is how you will be searching through the audit logs. Like roles updated or members banned.
+  // *A complete list of types can be found at the end of this page.*
+  
+  // Define an empty user for now.
+  let user = ""
+  const executor = await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(audit => audit.entries.first().executor.username)
+  
+  
+})
+```
+
+What about bots? Bot messages can be deleted, and they can delete thier own. How can we tell if it is a bot? Great question. The answer is you don't, at least not without some sort of storage data within your own bot. I used a small enmap database that allows me to show that the user was the bot based on where the bot is deleting the messages, whether it was a innapropriate word or a link. 
+
+Let's take a break to explain exactly whats going on in the below code block. The `Date.now()` is getting the current time (in milliseconds). We want to take away 5 seconds for the potential delay in the audit logs. The executor will be retrieving the very latest audit log entry and all of its information that goes with it. What does this information contain? Everything we need for logging. 
+
+```js
+GuildAuditLogsEntry {
+  targetType: 'MESSAGE',
+  actionType: 'DELETE',
+  action: 'MESSAGE_DELETE',
+  reason: null,
+  executor: [Object],
+  changes: null,
+  id: '434116792567201792',
+  extra: [Object],
+  target: [Object] }
+```
+Notice the `reason` field. Some audit logs, like kicking and banning, can provide a reason. You can probably make the bot log when a user is banned and for whatever reason. What we want is the executor of the action. We do that by going to the `executor` target as that is where the user is stored. Below is showing what the information that the `executor` property provides us:
+```js
+User {
+  id: '286270602820452353',
+  username: 'Zoth The Wumpus',
+  discriminator: '6066',
+  avatar: '57361ef0f8e23e02a44069c3dd5f5f47',
+  bot: false,
+  lastMessageID: '435165896198062091',
+  lastMessage: [Object] }
+```
+
+With all the information above, we can start creating comparisons to narrow down on who really deleted the message, whether it was the author of the message or someone else.
+
+```js
+  // Please keep in mind: Discord's audit logs will not log the information if the author of that message deleted it.
+  // I did this with a series of checks. 
+  // We are checking if the channel ID is the same as the id the message is deleted it
+  if ((await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().extra.channel.id === message.channel.id)) 
```

Comment:
Congrats, in this line and the next 9 line you've just made 7 absolutely useless requests to the API. Fetch the log *once* at the top of the script and then refer to that variable in all other instances. This is what I was talking about with "don't return the executor, return the entry and get its properties". Doing multiple requests is spammy, might break ratelimits, and will slow down this process to hell. 

```js
const entry = await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first());
```
Boom, done, now `entry` has all the things you need.